gwt-material/src/main/java/gwt/material/design/client/ui/MaterialCutOut.java

```diff
@@ -0,0 +1,297 @@
+
+    private String backgroundColor = "rgba(30,136,229,0.85)";
+    private int cutOutPadding = 10;
+    private boolean circle = false;
+    private boolean autoAddedToDocument = false;
+    private Element targetElement;
+    private Element focus;
+
+    public MaterialCutOut() {
+        super("");
+        focus = Document.get().createDivElement();
+        focus.setId(DOM.createUniqueId());
+        getElement().appendChild(focus);
+
+        setStyleName("material-cutout");
+        Style style = getElement().getStyle();
+        style.setWidth(100, Unit.PCT);
+        style.setHeight(100, Unit.PCT);
+        style.setPosition(Position.FIXED);
+        style.setTop(0, Unit.PX);
+        style.setLeft(0, Unit.PX);
+        style.setOverflow(Overflow.HIDDEN);
+        style.setZIndex(10000);
+        style.setDisplay(Display.NONE);
+
+        focus.setClassName("material-cutout-focus");
+        style = focus.getStyle();
+        style.setProperty("content", "\'\'");
+        style.setPosition(Position.ABSOLUTE);
+        style.setZIndex(-1);
+    }
+
+    /**
+     * Sets the raw CSS background color of the cut out. All accepted CSS colors
+     * can be used, such as:
+     * <ul>
+     * <li>Named colors: white, blue, red</li>
+     * <li>HEX colors: #fff, #1e88e5</li>
+     * <li>RGB colors: rgb(255,255,255), rgb(140,200,50)</li>
+     * <li>RGBA colors (recommended): rgb(255,255,255,0.70),
+     * rgba(30,136,229,0.85)</li>
+     * </ul>
+     */
+    public void setBackgroundColor(String backgroundColor) {
+        this.backgroundColor = backgroundColor;
+    }
+
+    /**
+     * @return The raw CSS background color
+     */
+    public String getBackgroundColor() {
+        return backgroundColor;
+    }
+
+    /**
+     * Sets if the cut out should be rendered as a circle or a simple rectangle.
+     * Circle is better for targets with same width and height. The default is
+     * <code>false</code> (is a rectangle).
+     */
+    public void setCircle(boolean circle) {
+        this.circle = circle;
+    }
+
+    /**
+     * @return The if the cut out should be rendered as a circle or a simple
+     *         rectangle
+     */
+    public boolean isCircle() {
+        return circle;
+    }
+
+    /**
+     * Sets the padding in pixels of the cut out focus in relation to the target
+     * element. The default is 10.
+     */
+    public void setCutOutPadding(int cutOutPadding) {
+        this.cutOutPadding = cutOutPadding;
+    }
+
+    /**
+     * @return The padding in pixels of the cut out focus in relation to the
+     *         target element
+     */
+    public int getCutOutPadding() {
+        return cutOutPadding;
+    }
+
+    /**
+     * Sets the target element to be focused by the cut out. To get the element
+     * from a Widget, you can use:
+     * 
+     * <pre>
+     * Element element = widget.getElement();
+     * </pre>
+     */
+    public void setTargetElement(Element targetElement) {
```

Comment:
instead of cutout.setTargetElement(Element);
can you make it like:
cutout.setTarget(Widget);
to make it more convenient to use :)